src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -336,8 +325,7 @@ private static bool HasActionsForEvent(CompilationEvent compilationEvent, Analyz
         public async Task OnCompilationEventProcessedAsync(CompilationEvent compilationEvent, ImmutableArray<DiagnosticAnalyzer> analyzers, Func<ISymbol, DiagnosticAnalyzer, Task> onSymbolAndMembersProcessedAsync)
         {
             // Analyze if the symbol and all its declaring syntax references are analyzed.
-            var symbolDeclaredEvent = compilationEvent as SymbolDeclaredCompilationEvent;
-            if (symbolDeclaredEvent != null)
```

Comment:
📝 This also was not needed for this change

Reply:
null-ness related changes are in scope of this PR

Reply:
The `var` already inferred the correct type, and flow analysis handled the `!= null`.